pepper-apis/src/main/java/org/broadinstitute/ddp/Housekeeping.java

```diff
@@ -253,160 +254,176 @@ public static void start(String[] args, SendGridSupplier sendGridSupplier) {
+                // in one transaction, query the list of events to consider dispatching
+                final List<QueuedEventDto> pendingEvents = new ArrayList<>();
+                TransactionWrapper.useTxn(TransactionWrapper.DB.APIS, apisHandle -> {
                     EventDao eventDao = apisHandle.attach(EventDao.class);
-                    JdbiQueuedEvent queuedEventDao = apisHandle.attach(JdbiQueuedEvent.class);
                     // first query the full list of pending events, shuffled to avoid event starvation
                     LOG.info("Querying pending events");
-                    List<QueuedEventDto> pendingEvents = eventDao.findPublishableQueuedEvents();
+                    pendingEvents.addAll(eventDao.findPublishableQueuedEvents());
                     LOG.info("Found {} events that may be publishable", pendingEvents.size());
                     Collections.shuffle(pendingEvents);
+                });
 
-                    for (QueuedEventDto pendingEvent : pendingEvents) {
+                for (QueuedEventDto pendingEvent : pendingEvents) {
+                    // for each event we are considering, handle it in its own transaction
+                    TransactionWrapper.useTxn(TransactionWrapper.DB.APIS, apisHandle -> {
+                        boolean shouldSkipEvent = false;
+                        JdbiQueuedEvent queuedEventDao = apisHandle.attach(JdbiQueuedEvent.class);
+                        EventDao eventDao = apisHandle.attach(EventDao.class);
                         if (pendingEvent.getParticipantGuid() != null) {
                             User participant = apisHandle.attach(UserDao.class)
                                     .findUserByGuid(pendingEvent.getParticipantGuid())
                                     .orElse(null);
                             if (participant == null) {
                                 LOG.error("Could not find participant {} for publishing queued event {}, skipping",
                                         pendingEvent.getParticipantGuid(), pendingEvent.getQueuedEventId());
-                                continue;
+                                shouldSkipEvent = true;
                             } else if (participant.isTemporary()) {
                                 LOG.warn("Participant {} for queued event {} is a temporary user, skipping",
                                         pendingEvent.getParticipantGuid(), pendingEvent.getQueuedEventId());
-                                continue;
+                                shouldSkipEvent = true;
                             }
                         }
 
-                        String pendingEventId = pendingEvent.getDdpEventId();
-                        boolean shouldCancel = false;
-                        if (StringUtils.isNotBlank(pendingEvent.getCancelCondition())) {
-                            try {
-                                shouldCancel = pexInterpreter.eval(pendingEvent.getCancelCondition(),
```

Comment:
The diffs here are misleading--things have moved around due to indents, but I don't believe I have really deleted anything.

Reply:
The diffs in intellij are much better--take a look there to get a better sense of what's changed.

Reply:
Github feature desperately needed: diff with ignore whitespace!